### ISSUE TITLE
Why does Unknown alert state fire a notification?

### ISSUE THREAD

**Issue by anotherfatcat** [grafana/grafana#18470]
**What happened**:
When an alert is in Unknown state, it triggers a notification.

**What you expected to happen**:
In https://grafana.com/docs/alerting/notifications/, it describes some valid states of an alert:
"state - The possible values for alert state are: ok, paused, alerting, pending, no_data."
If the state is changed to a value which is out of above list, it should not fire a notification.
If it is By Design to send a notification for changing state to Unknown, could you please share some clues about the time when this state will fire a notification and it will also be reminded or not?

**How to reproduce it (as minimally and precisely as possible)**:
In Conditions, add some fields which has no data at all.
Set state of condition "If no data or all values are null" to "Keep Last State". 
After this, if the fields happened to have some data and then no data again, it is possible to receive a notification and the state of this alert is Unknown.

**Anything else we need to know?**:
We used Kafka as our channel, and 

**Environment**:
- Grafana version:  6.2.4
- Data source type & version:  InfluxDB
- OS Grafana is installed on:  Ubuntu 18.04
- User OS & Browser: 
- Grafana plugins: No
- Others:

**Comment by anotherfatcat** [comment]
@marefr May I know if any information could be shared about this logic is By Design or a bug?
Thanks.

**Comment by marefr** [comment]
@anotherfatcat we need to investigate to be able to tell if it's a bug or by design, but we haven't had time to look into this yet.

**Comment by marefr** [comment]
We're still planning to investigate this

**Comment by marefr** [comment]
First of all, unknown is the initial state of an alert rule when it haven't been evaluated for the first time. I would guess since your using "If no data or all values are null" to "Keep Last State" there's some edge case where initial unknown state is the last state. But I've tried to reproduce this without success. 

One thing that wasn't really clear to me was when "if the fields happened to have some data" what state is alert rule transitioned to before "and then no data again?

I did test on the latest version. I would suggest you to test a newer version. I'll close this for now, but will reconsider reopen if you provide more details of how you managed to trigger this scenario.
